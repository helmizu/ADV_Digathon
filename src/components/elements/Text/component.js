import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

export default class Component extends React.Component {
  render() {
    const { children, classes, className, component = 'span', variant, inline, ...rest } = this.props;
    const customClass = className ? `${classes.text} ${className}` : classes.text;
    const props = {
      display: 'block',
      variant: 'body2'
    };
    if (inline) {
      props.display = 'inline';
    }
    if (variant) {
      props.variant = variant;
    }
    return (
      <Typography className={customClass} component={component} {...props} {...rest}>
        {children}
      </Typography>
    );
  }
}

Component.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object,
  className: PropTypes.string,
  component: PropTypes.node,
  inline: PropTypes.bool,
  variant: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
};
Component.defaultProps = {
  children: undefined,
  classes: {},
  className: '',
  component: undefined,
  inline: false,
  variant: false
};
