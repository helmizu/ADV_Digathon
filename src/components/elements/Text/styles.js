const styles = {
  text: {}
};

export default styles;
