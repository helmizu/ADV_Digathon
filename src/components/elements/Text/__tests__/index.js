import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../index';

jest.mock('react-router-dom');

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Text component="span">Text</Text>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
