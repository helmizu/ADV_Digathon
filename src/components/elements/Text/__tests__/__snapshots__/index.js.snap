// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<span
  className="MuiTypography-root Component-text-1 MuiTypography-body2 MuiTypography-displayBlock"
>
  Text
</span>
`;
